src/components/AddBook.jsx

```diff
@@ -1,17 +1,41 @@
-import React from 'react'
-import Navigation from './Navigation'
-import ApiSearch from './ApiSearch'
+import React, { useEffect } from "react";
+import Navigation from "./Navigation";
+import SearchBox from "./ui/SearchBox";
+import { useState } from "react";
+import BookDetails from "./BookDetails";
 
-const AddBook = () => {
-  return (
-    <div>
-      <Navigation/>
-        <br/>
-      <ApiSearch />
 
-    </div>
-    
-  )
-}
+const AddBook = () =>{
+    const [ isSearching, setIsSearching] = useState(true);
+    const [ isClicked, setIsClicked ] = useState(false);
+    const [clickedBook, setClickedBook] = useState(null);
 
-export default AddBook
\ No newline at end of file
+  const handleClickTwo = (obj) => {
+    setClickedBook(obj);
+    setIsClicked(true);
+    setIsSearching(false);
+  };
+   
+    if (isSearching) {
+    return (
+        < div>
+            <Navigation />
+            <br/>
+            <SearchBox handleClickTwo ={handleClickTwo} />
+        </div>
+    )
+    }
+
+
+
+    if(isClicked) {
+    return(
+        <div>
+            <Navigation />
+            <br/>
+            <BookDetails clickedBook={clickedBook}/>
+        </div>
+    )
+    }
```

Comment:
By writing the code this way, the "back" button does not work and the user will have to start a whole new search each time.  I think the buttom should navigate to a URL for the specific book.

Reply:
I did it this way for a couple of reasons. 
the book object returned by the API was super problematic. Often really large, missing significant chunks of data, etc. 
I originally wanted to pass the bookkey over as a query parameter and using React Query for a fresh API call but I could not get it working. On occasion I was even getting multiple results and I was having to truncate and add values because of how the bookkey.value shows up as /works/ at the start was problematic for react router due to the "/" involved. The book objects were already being formatted and truncated by search to make them easier to pass through props as well. 
Doing it this way allows the book to pass as props. 

The good news is this can be easily fixed. Would just need to add a button to the top left of the details page that simply reverts the state back and you will be back to search results.